kafka-connect-neo4j/docker/readme.adoc

```diff
@@ -339,6 +339,162 @@ In case of relationship
+If no prefix is specified this means that the property will be included
+
+*Note*: you cannot mix inclusion and exclusion so if your pattern must contains all exclusion
+or inclusion properties
+
+Labels can be chained via `:`
+
+====== The `Node Pattern` configuration
+
+You can configure the node pattern extraction in the following way:
+
+```
+streams.sink.topic.pattern.node.<TOPIC_NAME>=<NODE_EXTRACTION_PATTERN>
+```
+
+So for instance, given the following `json` published via the `user` topic:
+
+[source,json]
+----
+{"userId": 1, "name": "Andrea", surname:"Santurbano", "address": {"city": Venice, cap: "30100"}}
+----
+
+You can transform it into a node by providing the following configuration:
+
+
+by specifying a Cypher like node pattern:
+
+```
+streams.sink.topic.pattern.node.user=(:User{!userId})
+```
+
+or by specifying a simpler pattern:
+
+```
+streams.sink.topic.pattern.node.user=User{!userId}
+```
+
+Similar to the CDC pattern you can provide:
+
+[cols="1m,3a",opts=header]
+|===
+| pattern
+| meaning
+
+| User:Actor{!userId} or User:Actor{!userId,*}
+| the userId will be used as ID field and all properties of the json will be attached to the node with the provided
+labels (`User` and `Actor`) so the persisted node will be: *(User:Actor{userId: 1, name: 'Andrea', surname: 'Santurbano', `address.city`: 'Venice', `address.cap`: 30100})*
+
+| User{!userId, surname}
+| the userId will be used as ID field and *only* the surname property of the json will be attached to the node with the provided
+labels (`User`) so the persisted node will be: *(User:Actor{userId: 1, surname: 'Santurbano'})*
+
+| User{!userId, surname, address.city}
+| the userId will be used as ID field and *only* the surname and the `address.city` property of the json will be attached to the node with the provided
+labels (`User`) so the persisted node will be: *(User:Actor{userId: 1, surname: 'Santurbano', `address.city`: 'Venice'})*
+
+| User{!userId,-address}
+| the userId will be used as ID field and the `address` property will be excluded
+so the persisted node will be: *(User:Actor{userId: 1, name: 'Andrea', surname: 'Santurbano'})*
+
+|===
+
+====== The `Relationship Pattern` configuration
+
+You can configure the relationship pattern extraction in the following way:
+
+```
+streams.sink.topic.pattern.relationship.<TOPIC_NAME>=<RELATIONSHIP_EXTRACTION_PATTERN>
+```
+
+So for instance, given the following `json` published via the `user` topic:
+
+[source,json]
+----
+{"userId": 1, "productId": 100, price: 10, currency: "€", "shippingAddress": {"city": Venice, cap: "30100"}}
+----
+
+You can transform it into a node by providing the following configuration:
+
+By specifying a Cypher like node pattern:
+
+```
+streams.sink.topic.pattern.node.user=(:User{!userId})-[:BOUGHT{price, currency}]->(:Product{!productId})
+```
+
+or by specifying a simpler pattern:
+
+```
+streams.sink.topic.pattern.node.user=User{!userId} BOUGHT{price, currency} Product{!productId}
```

Comment:
we should show the simple patterns first.

Reply:
ok